docs/api/v3.rst

```diff
@@ -227,6 +227,35 @@ Project create
     :statuscode 400: Some field is invalid
 
 
+Project update
+++++++++++++++
+
+.. http:patch:: /api/v3/projects/(string:project_slug)/
+
+    Edit an existent project.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl \
+          -X PATCH \
+          -H "Authorization: Token <token>" https://readthedocs.org/api/v3/projects/pip/ \
+          -H "Content-Type: application/json" \
+          -d @body.json
+
+    The content of ``body.json`` is like,
+
+    .. sourcecode:: json
+
+        {
+            "name": "New name for the project",
+            "default_version": "v0.27.0"
+        }
+
+    :statuscode 204: Updated successfully
```

Comment:
I assume it will also raise an error in some cases?

Reply:
On the "Ship APIv3" PR I was thinking to remove all the redundant docs or un-useful list of fields: avoid things like `slug: the slug of the Project` and just document that ones that are useful to avoid.

I applied the same concept for status codes, mentioning only the important ones (based on David's comment as well: https://github.com/readthedocs/readthedocs.org/pull/4863#discussion_r230834643)

I did a commit for this in that PR: https://github.com/readthedocs/readthedocs.org/pull/6169/commits/f7e168f929bff3a33b48e066786cfada653536ee

What do you think?

Reply:
Seems reasonable 👍